nobl9/provider.go

```diff
@@ -134,7 +134,19 @@ func getClient(config ProviderConfig, project string) (*nobl9.Client, diag.Diagn
 			config.OktaOrgURL,
 			config.OktaAuthServer,
 		)
+	}
+
+	once.Do(func() {
+		clients = make(map[string]*nobl9.Client)
+		clients[project], clientErr = newClient()
 	})
+
+	client, clientInitialized := clients[project]
+	if !clientInitialized {
+		client, clientErr = newClient()
+		clients[project] = client
```

Comment:
this initialization will is not thread safe - we might get into concurrent map write here.